### ISSUE TITLE
Custom Go-lang types for DB models (Monetary values)

### ISSUE THREAD

**Issue by ayushajain** [go-jet/jet#56]
We are currently storing monetary values as type Numeric in our PostgreSQL database and we perform all types of numeric operations. Because of this, the default golang Float64 datatype does not work for our needs as we cannot work with numerical approximations.

Is there any way to customize the model currently?

**Comment by go-jet** [comment]
It is not possible to customize generated model at the moment. This feature will be added later. 

But it is possible to create custom model file, and even to combine custom with generated model files.
Also as a go struct field it is possible to use any type that implements `Scanner` interface.

Assuming table name is `my_table`, column is `money` and that you want to retrieve sql numeric type to go integer type, something like this should be possible:

```go
type MoneyType int64  // or some other type 

func (e *MoneyType) Scan(value interface{}) error {
 ...  add implementation
}

type MyTable struct {
    model.MyTable
    Money MoneyType
}
```

Now it is possible to use `MyTable` in any place where `model.MyTable` is used before.

**Comment by mlaflamm** [comment]
We ran into a similar issue. Unfortunately even with your suggestion to use a customised model, the value is converted to a `sql.NullFloat64` in [qrm/utill.go](https://github.com/go-jet/jet/blob/master/qrm/utill.go#L311-L312) before being passed to the custom type `Scan`. 

To work correctly, the value must be converted to a lossless decimal type, like [ericlagergren/decimal](https://github.com/ericlagergren/decimal). Using `sql.NullString` instead of `sql.NullFloat64` would likely work too. I am sure this could be implemented in a way that doesn't break projects that use the generated models with `float64`.

**Comment by go-jet** [comment]
Yes, you right, exact decimal types(`decimal` and `numeric`) should have been first converted to string.  
I've made a quick fix([decimals](https://github.com/go-jet/jet/tree/decimals) branch) which enables scan into custom lossless decimal types using customized model types.([sample usage](https://github.com/go-jet/jet/commit/cc18ba210c01434eb39866f7a55af1d541746c78#diff-d120a6b2e75297a13b3038b7cde26876f93b3e406d958c99e0ac1e0f5cd3b513R48))
_Note that scan from `decimal` and `numeric` into float64 is disabled on this branch._

**Comment by mlaflamm** [comment]
Would it be possible to base your branch from `develop`? We would like to try it but we have a strong dependency on #51.

**Comment by crazydadz** [comment]
I've done a PR (#61) based on your decimal branch that is backward compatible with existing generated models that uses the float64 type. All the postgres tests are passing. This may breaks custom models that defined their own decimal type but paradoxically this may be a good thing if they cares about decimal precision.

**Comment by go-jet** [comment]
@crazydadz Agreed, make sense.
@mlaflamm Rebased.

**Comment by crazydadz** [comment]
I've created a new PR that add a test to write a decimal into the database. #62

**Comment by mlaflamm** [comment]
@go-jet thank you for making this change. We now use the `decimal` branch as our _officially supported_ jet version. It would be great if you merge that branch into `develop` and eventually do an official release. We decided to use `string` in our models instead of a third party decimal library. We do the appropriate conversion elsewhere as needed. 

FYI we use a bash script wrapper around the `jet` code generator to replace the types in the models when they are generated. I documented what we do in this gist:
https://gist.github.com/mlaflamm/5093c5877b401f608a1a737fab7b662c

This might be of interest for other jet users.
